docs(yaml): add section about yaml anchor (#4203)
Mergifyio/mergify-engine: docs/source/configuration.rst

   1 d 15 h 6 m 42 s

> YAML anchors and aliases

Reviewer: ```suggestion
YAML Anchors and Aliases
```

~~~~~~~~~~~~~~~~~~~~~~~~

The configuration file supports `YAML anchors and aliases <https://yaml.org/spec/1.2.2/#anchors-and-aliases>`.

> It allows to reuse configuration section. You can reused the continuous integration lists for example:

Reviewer: ```suggestion
It allows reusing configuration sections. For example, you could reuse the list of continuous integration checks:
```
